app/src/main/java/com/example/homew3/mvvm/view/RecipesFragment.kt

```diff
@@ -0,0 +1,127 @@
+
+    private var _binding: FragmentRecipesBinding? = null
+    private val binding get() = requireNotNull(_binding)
+
+    private val adapter by lazy {
+        RecipesAdapter(
+            context = requireContext(),
+            onRecipeClicked = {
+                findNavController().navigate(
+                    RecipesFragmentDirections.toMoreInfoFragment(
+                        it.id,
+                        it.title,
+                        it.image
+                    )
+                )
+            }
+        )
+    }
+
+    private val recipesDao by lazy {
+        requireContext().appDataBase.recipesDao
+    }
+
+    private val viewModel: RecipesViewModel by viewModels(
+        factoryProducer = {
+            viewModelFactory {
+                initializer {
+                    RecipesViewModel(ServiceLocator.provideRecipes(), recipesDao)
+                }
+            }
+        }
+    )
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentRecipesBinding.inflate(inflater, container, false)
+            .also { _binding = it }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(binding) {
+
+            recyclerView.adapter = adapter
+
+            recyclerView.addItemDecoration(
+                object : RecyclerView.ItemDecoration() {
+                    override fun getItemOffsets(
+                        outRect: Rect,
+                        view: View,
+                        parent: RecyclerView,
+                        state: RecyclerView.State
+                    ) {
+                        outRect.bottom = 50   //R.dimen.default_padding если делаю так, то только один объект отображется =(
+                    }
+                }
+            )
+
+            swipeRefresh.setOnRefreshListener {
+                adapter.submitList(emptyList())
+                viewModel.onRefreshedRecipes()
+            }
+
+
+            toolbar.menu
+                .findItem(R.id.action_search)
+                .actionView
+                .let { it as SearchView }
+                .setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+                    override fun onQueryTextSubmit(query: String): Boolean {
+                        return false
+                    }
+
+                    override fun onQueryTextChange(query: String): Boolean {
+                        viewModel.onQueryChanged(query)
+                        return true
+                    }
+                })
+
+
+            viewModel
+                .recipeFlow
+                .flowWithLifecycle(viewLifecycleOwner.lifecycle, Lifecycle.State.STARTED)
+                .onEach { swipeRefresh.isRefreshing = false }
+                .onEach { adapter.submitList(it) }
```

Comment:
можно ссылкой на метод 
```suggestion
                .onEach(adapter::submitList)
```